backend-shared/config/db/util/utils_test.go

```diff
@@ -1,12 +1,881 @@
+
+			isNew, err = GetOrCreateDeploymentToApplicationMapping(ctx, deploymentToApplicationMapping, dbQueries, log)
+			Expect(err).To(BeNil())
+			Expect(isNew).To(BeFalse())
+		})
+
+		It("Should delete existing DeploymentToApplicationMapping, if a resource is passed having same name/namespace, so there is be only one resource per name and namespace.", func() {
+
+			// ----------------------------------------------------------------------------
+			By("Create first DeploymentToApplicationMapping resource.")
+			// ----------------------------------------------------------------------------
+
+			isNew, err := GetOrCreateDeploymentToApplicationMapping(ctx, deploymentToApplicationMapping, dbQueries, log)
+			Expect(err).To(BeNil())
+			Expect(isNew).To(BeTrue())
+
+			// ----------------------------------------------------------------------------
+			By("Create second DeploymentToApplicationMapping resource, having same name and namespace as first, but different ID.")
+			// ----------------------------------------------------------------------------
+
+			deploymentToApplicationMappingSecond := &db.DeploymentToApplicationMapping{
+				Deploymenttoapplicationmapping_uid_id: string(uuid.NewUUID()),
+				Application_id:                        application.Application_id,
+				DeploymentName:                        "my-depl-to-app-mapping",
+				DeploymentNamespace:                   "my-namespace",
+				NamespaceUID:                          "test-1",
+			}
+
+			isNew, err = GetOrCreateDeploymentToApplicationMapping(ctx, deploymentToApplicationMappingSecond, dbQueries, log)
+			Expect(err).To(BeNil())
+			Expect(isNew).To(BeTrue())
+
+			// ----------------------------------------------------------------------------
+			By("Check first DeploymentToApplicationMapping is deleted in DB.")
+			// ----------------------------------------------------------------------------
+
+			err = dbQueries.GetDeploymentToApplicationMappingByDeplId(ctx, &db.DeploymentToApplicationMapping{
+				Deploymenttoapplicationmapping_uid_id: deploymentToApplicationMapping.Deploymenttoapplicationmapping_uid_id,
+			})
+			Expect(err).NotTo(BeNil())
+
+			// ----------------------------------------------------------------------------
+			By("Check second DeploymentToApplicationMapping resource is created in DB.")
+			// ----------------------------------------------------------------------------
+
+			err = dbQueries.GetDeploymentToApplicationMappingByDeplId(ctx, &db.DeploymentToApplicationMapping{
+				Deploymenttoapplicationmapping_uid_id: deploymentToApplicationMappingSecond.Deploymenttoapplicationmapping_uid_id,
+			})
+			Expect(err).To(BeNil())
+
+			// send second resource object to get deleted in AfterEach
+			deploymentToApplicationMapping = deploymentToApplicationMappingSecond
+		})
+	})
+
+	Context("Testing for GetOrCreateGitopsEngineClusterByKubeSystemNamespaceUID function.", func() {
+
```

Comment:
It can be tough to tell whether the story criteria items are being met when it comes to writing tests, due to the nature of test writing.

In order to tell if a particular part of a test is being covered, you can verify this by running `make test`, which will produce a coverage.out file. Then, in VS Code, select `View` > `Command Palette`, then `Go: Apply Cover Profile`, and specify the absolute path of the coverage.out file.

In this case, after doing that, it appears that the following story criteria items are not being fully tested:
- if a KubernetesToDBResourceMapping mapping exists, and the GitOpsEngineCluster row doesn't exist: The GetOrCreateGitopsEngineClusterByKubeSystemNamespaceUID function should delete the old mapping, and create a new mapping/cluster
- if a KubernetesToDBResourceMapping mapping exists, and the GitOpsEngineCluster row exists: The  GetOrCreateGitopsEngineClusterByKubeSystemNamespaceUID should not create, and should just retrieve existing values from the database

You you follow the above steps, then open `GetOrCreateGitopsEngineClusterByKubeSystemNamespaceUID` function, you will see that 'Scenario C' and 'Scenario D' code blocks are not covered. This corresponds to the two test items above. (Don't worry about 'Scenario B'.) You can verify using `Go: Apply Cover Profile`, as above.

(https://issues.redhat.com/browse/GITOPSRVCE-134)